### DISCUSSION TITLE
Distributed Deployment Pattern via Kubernetes

### DISCUSSION THREAD

**Discussion by muscles18** [microsoft/autogen#441] [Ideas]
Exciting times in AI.  Wondering if there is a way to organize this framework to be deployable patterns in a cluster environment? The example I am thinking of would be to allow users to establish resource definitions on a Kubernetes cluster such that I can spin a unique agent as pod and essentially allow each pod to enter into conversations as orchestrated by a cluster orchestrator

**Comment by sonichi** [comment]
Interesting idea. This thread is worth discussion. It's an upscaled version of #143

**Reply by muscles18** [reply]
My thinking is that this would allow for fault tolerance and horizontal scaling where you could establish pools of personas who could be pulled in by the cluster orchestrator to complete tasks.  If deployed as TeachableAgents could continue to improve at their tasks over time and this sort of lay the groudwork for a true AI workforce that would be self improving
